editor/src/components/editor/actions/actions.tsx

```diff
@@ -3094,64 +3098,125 @@ export const UPDATE_FNS = {
+          }
+          return {
+            ...e,
+            pasteTargetsToIgnore: e.selectedViews,
+            internalClipboard: {
+              styleClipboard: [],
+              elements: copyData?.data ?? [],
+            },
+          }
+        },
+        dispatch,
+      )
+    }
+
+    const canReparent = sequenceEither(
+      editor.selectedViews.map((target) => canCopyElement(editor, target)),
     )
+
+    if (isLeft(canReparent)) {
+      const showToastAction = showToast(notice(canReparent.value))
+      return UPDATE_FNS.ADD_TOAST(showToastAction, editor)
+    }
+
+    const copyData = createClipboardDataFromSelection(editor, builtInDependencies)
+    if (copyData != null) {
+      // side effect 😟
+      Clipboard.setClipboardData({
+        plainText: copyData.plaintext,
+        html: encodeUtopiaDataToHtml(copyData.data),
+      })
+    }
+
+    return {
+      ...editor,
+      pasteTargetsToIgnore: editor.selectedViews,
+      internalClipboard: {
+        styleClipboard: [],
+        elements: copyData?.data ?? [],
+      },
+    }
   },
   CUT_SELECTION_TO_CLIPBOARD: (
-    editorForAction: EditorModel,
+    editor: EditorModel,
     dispatch: EditorDispatch,
     builtInDependencies: BuiltInDependencies,
   ): EditorModel => {
-    return toastOnUncopyableElementsSelected(
-      'Cannot cut these elements.',
-      editorForAction,
-      false,
-      (editor) => {
-        // side effect 😟
-        const copyData = createClipboardDataFromSelection(editorForAction, builtInDependencies)
-        if (copyData != null) {
-          Clipboard.setClipboardData({
-            plainText: copyData.plaintext,
-            html: encodeUtopiaDataToHtml(copyData.data),
-          })
-        }
-        return UPDATE_FNS.DELETE_SELECTED(
-          {
-            ...editor,
-            pasteTargetsToIgnore: editor.selectedViews,
-            internalClipboard: {
-              styleClipboard: [],
-              elements: copyData?.data ?? [],
+    if (!isFeatureEnabled('Paste with props replaced')) {
+      return toastOnUncopyableElementsSelected(
+        'Cannot cut these elements.',
+        editor,
+        false,
+        (e) => {
+          // side effect 😟
+          const copyData = createClipboardDataFromSelection(e, builtInDependencies)
+          if (copyData != null) {
+            Clipboard.setClipboardData({
+              plainText: copyData.plaintext,
+              html: encodeUtopiaDataToHtml(copyData.data),
+            })
+          }
+          return UPDATE_FNS.DELETE_SELECTED(
+            {
+              ...e,
+              pasteTargetsToIgnore: e.selectedViews,
+              internalClipboard: {
+                styleClipboard: [],
+                elements: copyData?.data ?? [],
+              },
             },
-          },
-          dispatch,
-        )
+            dispatch,
+          )
+        },
+        dispatch,
+      )
+    }
+
+    const canReparent = sequenceEither(
```

Comment:
Is this 15-ish lines the same as the block above?

Reply:
I was very much on the fence about this too, the reason I left in this duplication is that there isn't that much going on here:
- checking if an element can be copied + early return
- updating the clipboard
- returning the editor state or deleting the cut elements
The third step already can't be generalised among copy and cut, so I decided to leave the code as it is, and when the FS is cleaned up, clean up this part too.

Reply:
Yeah, we can totally refactor this later on as an atomic thing, as opposed to it getting threaded into the work you're doing.

Reply:
OK, looking at @Rheeseyb 's comment, this isn't negotiable 😅

Reply:
Yeah, the duplication shouldn't block this PR, since to tidy it up requires refactoring both action handlers. That can be tackled separately.

Reply:
I generalized some of the logic here [b71d842](https://github.com/concrete-utopia/utopia/pull/3778/commits/b71d8427d91cb1928b984f06c0456716f0ad5323)